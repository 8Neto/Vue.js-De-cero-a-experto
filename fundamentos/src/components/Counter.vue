<template>
    <h2>Counter</h2>
    <p>{{ counter }} <sup>2</sup> = {{ squareCounter }}</p>
</template>

<script>
export default {
    data() {
        return {
            counter: 10
        }
    },

    methods: {
        getSquareValue() {
            console.log('getSquareValue') // se ejecuta las veces que es llamada
            return this.counter * this.counter
        }
    },
    computed:{
        squareCounter() {
            console.log('squareCounter') //  se ejecuta solo una vez
            return this.counter * this.counter
        }
    },
}
</script>

<style scoped>
</style>