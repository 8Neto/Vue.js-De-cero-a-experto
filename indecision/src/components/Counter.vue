<template>
    <h2>{{ customTitle }}</h2>
    <p>{{ counter }} <sup>2</sup> = {{ squareCounter }}</p>

    <div>
        <!-- v-on: == @ -->
        <button @click="increase">+1</button>
        <button v-on:click="decrease">-1</button>
    </div>

</template>

<script>
export default {
    props: {
        title:String,
        start: {
            type: Number,
            default: 10,
            validator(value){
                return value > 0
            }
            // required: true

        }
    },
    data() {
        return {
            counter: this.start
        }
    },

    methods: {
        getSquareValue() {
            console.log('getSquareValue') // se ejecuta las veces que es llamada
            return this.counter * this.counter
        },
        increase(){
            this.counter++
        },
        decrease(){
            this.counter--
        },
    },
    computed:{
        squareCounter() {
            console.log('squareCounter') //  se ejecuta solo una vez
            return this.counter * this.counter
        },
        customTitle(){
            return this.title || 'Counter'
        }
    },
}
</script>

<style scoped>

    button {
        background-color: #64bb87;
        border-radius: 5px;
        border: 1px solid white;
        color: white;
        cursor: pointer;
        margin: 0 5px;
        padding: 5px 15px;
        transition: 0.3s ease-in-out;
    }

    button:hover {
        background-color: #5aa67b;
        transition: 0.3s ease-in-out;
    }

</style>