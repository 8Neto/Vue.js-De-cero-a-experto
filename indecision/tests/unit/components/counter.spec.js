/**
* @vitest-environment happy-dom
*/

import { mount, shallowMount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import Counter from '@/components/Counter.vue'


describe('Counter Component', ()=>{
    it('debe de hacer match con el snapshot', () =>{

        const wrapper = mount( Counter )

        expect(wrapper.html()).toMatchSnapshot()
    })
})
