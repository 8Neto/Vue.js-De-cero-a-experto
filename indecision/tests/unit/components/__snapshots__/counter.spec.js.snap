// Vitest Snapshot v1

exports[`Counter Component > debe de hacer match con el snapshot 1`] = `
"<h2 data-v-9bd91872=\\"\\">Counter</h2>
<p data-v-9bd91872=\\"\\">10 <sup data-v-9bd91872=\\"\\">2</sup> = 100</p>
<div data-v-9bd91872=\\"\\">
  <!-- v-on: == @ --><button data-v-9bd91872=\\"\\">+1</button><button data-v-9bd91872=\\"\\">-1</button>
</div>"
`;
